uber/models.py

```diff
@@ -388,17 +392,31 @@ def apply(self, params, *, bools=(), checkgroups=(), restricted=True, ignore_csr
 
                 try:
                     if isinstance(column.type, Float):
-                        value = float(value)
-                    elif isinstance(column.type, Choice) and value == '':
-                        value = None
+                        if value == '':
+                            value = None
+                        else:
+                            value = float(value)
                     elif isinstance(column.type, (Choice, Integer)):
-                        value = int(float(value))
+                        if value == '':
+                            value = None
+                        else:
+                            value = int(float(value))
                     elif isinstance(column.type, UTCDateTime):
-                        value = c.EVENT_TIMEZONE.localize(datetime.strptime(value, c.TIMESTAMP_FORMAT))
+                        try:
+                            value = datetime.strptime(value, c.TIMESTAMP_FORMAT)
```

Comment:
Everything in this entire function should really be handled inside the custom data type (like this should be part of `UTCDateTime` internals).

Aside from that, this code is much more liberal in what we accept as a date time. If parsing using `c.TIMESTAMP_FORMAT` fails, we fall back to `dateparser.parse(value)`, which will parse damn near anything.